src/Overlay.js

```diff
@@ -19,18 +19,14 @@ class OverlayRaw extends React.Component {
     if (Object.getOwnPropertyNames(this.props.Combatant).length === 0)
       return false
 
-    let maxRows = this.props.config.maxCombatants
-    let dataArray = Object.keys(this.props.Combatant)
-    let battler = dataArray.slice(0, maxRows)
-    let combatant
+    const maxRows = this.props.config.maxCombatants //not sure why let was being used here
+    const dataArray = this.props.isWebSocket ? Object.keys(this.props.Combatant).reverse() : Object.keys(this.props.Combatant) //const is fine, actws flips order for some reason I don't know why, this is my 'fix'
     let discordData = []
+	let currentRow = 0
 
-    for (const ref in battler) {
-      combatant = this.props.Combatant[battler[ref]]
-
-      // Send to Discord the right name in Settings
-      if (combatant.name.toUpperCase() === 'YOU')
-        combatant.name = this.props.config.characterName
+    //NOTE: instead of wasting time slicing the array up we just break the loop when we need to, it avoids the problem of filtering out the lb only to want to handle it later
+    for (const battler of dataArray) {
+      const combatant = this.props.Combatant[battler] //scope means we can just use const here
```

Comment:
I thank you for explaining your steps, I know it may look weird in some places but I never refactored since v1 😉 

Could you then remove the comments? They won't make sense after the PR is accepted.

Reply:
yeah I will remove the comments I'm busy with savage raids and life at the moment but I'll get round to it as soon as possible